frontend/src/contexts/players.tsx

```diff
@@ -18,21 +18,13 @@ type Context = {
   incrementPlayerScore: (playerId: string) => void;
 };
 
-const PlayersContext = React.createContext<Context>({
-  host: "",
-  setHost: () => null,
-  players: [],
-  initialisePlayers: () => null,
-  addPlayer: () => null,
-  removePlayer: () => null,
-  incrementPlayerScore: () => null,
-});
+const PlayersContext = React.createContext<PlayersContextType | undefined>(
+  undefined
```

Comment:
Why do these initialisation fields get removed? Because you've added memoisation?

Reply:
This change is related to the custom `usePlayers()` hook. Initialising the context object with `undefined` allows us to check this in the hook & ensure the hook is used as intended within the scope of a `PlayersProvider`.

Previously because the context is initialised with empty values and functions it would have the potential to confuse a developer trying to use the context *outside* of a provider as everything would appear initialised (players array, defined functions) however as those functions are defined `() => null` none of the functionality would work as expected. This change makes it very clear to developers that they're using it wrong by throwing an error in the `usePlayers()` hook